packages/assets-controllers/src/TokenListController.ts

```diff
@@ -143,40 +141,30 @@ export class TokenListController extends BaseControllerV2<
     this.updatePreventPollingOnNetworkRestart(preventPollingOnNetworkRestart);
     this.abortController = new WhatwgAbortController();
     if (onNetworkStateChange) {
-      onNetworkStateChange(async (networkStateOrProviderConfig) => {
-        // this check for "provider" is for testing purposes, since in the extension this callback will receive
-        // an object typed as NetworkState but within repo we can only simulate as if the callback receives an
-        // object typed as ProviderConfig
-        if ('providerConfig' in networkStateOrProviderConfig) {
-          await this.#onNetworkStateChangeCallback(
-            networkStateOrProviderConfig.providerConfig,
-          );
-        } else {
-          await this.#onNetworkStateChangeCallback(
-            networkStateOrProviderConfig,
-          );
-        }
+      onNetworkStateChange(async (networkControllerState) => {
+        await this.#onNetworkControllerStateChange(networkControllerState);
```

Comment:
We _can_ simulate a network controller state object in tests, so we don't need to make this check.